web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/DatastreamController.java

```diff
@@ -146,7 +154,24 @@ public void getThumbnail(@PathVariable("pid") String pidString,
             }
         }
 
-        derivativeContentService.streamData(pid, thumbName, principals, false, response);
+        accessControlService.assertHasAccess("Insufficient permissions to get thumbnail for " + pidString,
```

Comment:
I know this is the same point where it previous did the access check, but i think we should move it up to above the solr query, otherwise we're giving information about objects that the user doesn't have access to.

Reply:
ah yeah, i wasn't sure if the permissions check should be for the work or the file that is the thumbnail. i put it down here to check for the file but you're saying it should be for the work (if it's a work with a file) right?

Reply:
hmm, good point. It might be that we should be checking for both the work and the file